### PR TITLE
Remove code tagged with `@deprecated` 

### LINKED ISSUE #8567: Remove code tagged with `@deprecated`

### PR DESCRIPTION
**Pull request by tsteur**
fix #8567 

Didn't remove all deprecated methods but many. Some I removed the deprecated flag or added a note they should be kept in the code (eg `API.getPiwikVersion`). Some of the remaining ones we will remove maybe in Matomo 5 or so. To be seen. 

Before the first Matomo 4 beta or RC we'll need to sort the CHANGELOG file :)

### CODE REVIEW

File: core/ArchiveProcessor/PluginsArchiver.php [hunk 1]
```diff
@@ -81,11 +81,6 @@ public function __construct(Parameters $params, ArchiveWriter $archiveWriter = n
          *
          * @param bool $shouldAggregateFromRawData  Set to true, to aggregate from raw data, or false to aggregate multiple reports.
          * @param Parameters $params
-         * @ignore
```
Comment (tsteur) [inline core/ArchiveProcessor/PluginsArchiver.php:84]: still good to have this event

File: core/Plugin.php [hunk 2]
```diff
@@ -184,7 +184,6 @@ private function hasDefinedPluginInformationInPluginClass()
      * - 'theme' => bool                // Whether this plugin is a theme (a theme is a plugin, but a plugin is not necessarily a theme)
      *
      * @return array
-     * @deprecated
```
Comment (tsteur) [inline core/Plugin.php:187]: not sure why this was deprecated but seems useful

File: core/Plugin/API.php [hunk 3]
```diff
@@ -77,7 +77,7 @@ public static function getInstance()
     /**
      * Used in tests only
      * @ignore
-     * @deprecated
+     * @internal
```
Comment (tsteur) [inline core/Plugin/API.php:80]: same here seems useful... however only for internal usage. deprecated flag was maybe set so it doesn't appear in the developer docs

File: core/Site.php [hunk 4]
```diff
@@ -172,7 +172,6 @@ public static function setSiteFromArray($idSite, $infoSite)
      *                             array('idsite' => '2', 'name' => 'Site 2', ...),
      *                         )
      * @return array The modified array.
-     * @deprecated
```
Comment (tsteur) [inline core/Site.php:175]: still needed but not as api

File: plugins/API/API.php [hunk 5]
```diff
@@ -240,56 +227,6 @@ protected function getSegmentValuesFromVisitorLog($segmentName, $table)
...
-    }
-
-    /**
-     * Returns the URL to application SVG Logo
-     *
-     * @ignore
-     * @param bool $pathOnly If true, returns path relative to doc root. Otherwise, returns a URL.
-     * @return string
-     */
-    public function getSVGLogoUrl($pathOnly = false)
```
Comment (tsteur) [inline plugins/API/API.php:276]: the first 2 were marked as deprecated so figured I remove all logo related api methods

File: plugins/ExampleAPI/API.php [hunk 6]
```diff
@@ -32,16 +32,6 @@ public function getMatomoVersion()
         return Version::VERSION;
     }
 
-    /**
-     * Get Matomo version
-     * @return string
-     * @deprecated
-     */
-    public function getPiwikVersion()
```
Comment (tsteur) [inline plugins/ExampleAPI/API.php:40]: the example plugin is removed in the release anyway so this is safe to remove

File: plugins/Login/SessionInitializer.php [hunk 7]
```diff
@@ -21,11 +21,6 @@
 use Piwik\Session;
 
 /**
- * This SessionInitializer is no longer used, but is kept for backwards compatibility.
- * Session management no longer uses the piwik_auth cookie.
- *
- * @deprecated
- * @api
```
Comment (tsteur) [inline plugins/Login/SessionInitializer.php:28]: the class is still used in the login class but we no longer consider it as API

File: plugins/VisitsSummary/Controller.php [hunk 8]
```diff
@@ -40,22 +40,22 @@ public function __construct(Translator $translator)
     /**
      * @deprecated used to be a widgetized URL. There to not break widget URLs
      */
-    public function getSparklines()
+    public function index()
```
Comment (tsteur) [inline plugins/VisitsSummary/Controller.php:43]: had them first removed but be good to keep the widget urls working when it's that easy

File: core/Container/StaticContainer.php [hunk 9]
```diff
@@ -13,7 +13,8 @@
 /**
  * This class provides a static access to the container.
  *
- * @deprecated This class is introduced only to keep BC with the current static architecture. It will be removed in 3.0.
+ * @internal
```
Comment (tolbon) [inline core/Container/StaticContainer.php:16]: I was just wondering how to delete all instances of code that depended on it, do you have any idea or an example of good migrations StaticContainer to another things ?

Comment (sgiehl) [reply core/Container/StaticContainer.php:16]: I guess the solution would be using dependency injection everywhere. We already have that in some places, but it's not yet usable everywhere...

File: tests/PHPUnit/Integration/Columns/DimensionTest.php [hunk 10]
```diff
@@ -278,14 +278,6 @@ public function test_getId_ShouldCorrectlyGenerateIdFromDimensionsQualifiedClass
         $this->assertEquals("Test.DimensionTest", $this->dimension->getId());
     }
 
-    public function test_factory_ShouldCreateDimensionFromDimensionId()
-    {
-        Manager::getInstance()->loadPlugins(array('ExampleTracker'));
-
-        $dimension = Dimension::factory("ExampleTracker.ExampleDimension");
```
Comment (sgiehl) [inline tests/PHPUnit/Integration/Columns/DimensionTest.php:285]: Might be useful to transform that into a test of `DimensionsProvider::factory` instead, as that one seems untested.

**Review by sgiehl — CHANGES_REQUESTED** [review]

File: tests/PHPUnit/Integration/Columns/DimensionTest.php (hunk 10 above)
Comment (tsteur) [reply tests/PHPUnit/Integration/Columns/DimensionTest.php:285]: Done 👍

**Review by sgiehl — APPROVED** [review]

**Pull request merged**